frontend/packages/knative-plugin/src/topology/components/nodes/EventingPubSubNode.tsx

```diff
@@ -0,0 +1,147 @@
+import * as React from 'react';
+import * as classNames from 'classnames';
+import { Tooltip } from '@patternfly/react-core';
+import { useAccessReview } from '@console/internal/components/utils';
+import { modelFor, referenceFor } from '@console/internal/module/k8s';
+import {
+  Node,
+  observer,
+  useHover,
+  WithSelectionProps,
+  WithDndDropProps,
+  WithContextMenuProps,
+  useSvgAnchor,
+  useDragNode,
+  useCombineRefs,
+  WithCreateConnectorProps,
+  createSvgIdUrl,
+} from '@console/topology';
+import SvgBoxedText from '@console/dev-console/src/components/svg/SvgBoxedText';
+import {
+  NodeShadows,
+  NODE_SHADOW_FILTER_ID_HOVER,
+  NODE_SHADOW_FILTER_ID,
+  useSearchFilter,
+  useDisplayFilters,
+  nodeDragSourceSpec,
+  getTopologyResourceObject,
+  getFilterById,
+  SHOW_LABELS_FILTER_ID,
+} from '@console/dev-console/src/components/topology';
+import { TYPE_EVENT_PUB_SUB } from '../../const';
+
+import './EventingPubSubNode.scss';
+
+export type EventingPubSubNodeProps = {
+  element: Node;
+  canDrop?: boolean;
+  dropTarget?: boolean;
+  edgeDragging?: boolean;
+} & WithSelectionProps &
+  WithDndDropProps &
+  WithContextMenuProps &
+  WithCreateConnectorProps;
+
+const EventingPubSubNode: React.FC<EventingPubSubNodeProps> = ({
+  element,
+  canDrop,
+  dropTarget,
+  selected,
+  onSelect,
+  onContextMenu,
+  contextMenuOpen,
+  dndDropRef,
+  edgeDragging,
+  onHideCreateConnector,
+  onShowCreateConnector,
+}) => {
+  const svgAnchorRef = useSvgAnchor();
+  const [hover, hoverRef] = useHover();
+
+  const resourceObj = getTopologyResourceObject(element.getData());
+  const resourceModel = modelFor(referenceFor(resourceObj));
+  const editAccess = useAccessReview({
+    group: resourceModel.apiGroup,
+    verb: 'patch',
+    resource: resourceModel.plural,
+    name: resourceObj.metadata.name,
+    namespace: resourceObj.metadata.namespace,
+  });
```

Comment:
withEditReviewAccess HOC is used in componentFactory, so this component props will aleady have `editAccess` in its props, so we can remove this.